### PR TITLE
Added logic for the hp

### PR DESCRIPTION
**Pull request by sashadmii**
Added logic and styles to components:
- searchBar
- Cuisines
- List
- Footer

### CODE REVIEW

File: src/components/cuisines/Cuisines.jsx [hunk 1]
```diff
@@ -1,5 +1,38 @@
+import { useDispatch } from 'react-redux';
+
+import { changeList } from '../../store/list/listSlice';
+import { cuisines } from '../../store/list/constants';
+
 function Cuisines() {
-    return <aside>Cuisines</aside>;
+    const dispatch = useDispatch();
+
+    const handleClick = (cuisine) => {
```
Comment (kusochek) [inline src/components/cuisines/Cuisines.jsx]: rename function to fetchCuisine 
Or something similar

File: src/components/footer/Footer.jsx [hunk 2]
```diff
@@ -1,5 +1,40 @@
...
+
+            <div className="sm:text-center lg:text-end lg:w-1/4 ">
+                <h3 className="font-serifFont text-3xl mb-2">Contact Me:</h3>
+                <div className="flex gap-5 sm:justify-center lg:justify-end">
+                    {socialLinks.map((link) => (
+                        <a
+                            href={link.link}
+                            target="_blank"
+                            rel="noreferrer"
+                            key={link.name}>
```
Comment (kusochek) [inline src/components/footer/Footer.jsx:29]: Move > to the new line
And follow to this logic in other places

File: src/components/footer/Footer.jsx [hunk 3]
```diff
@@ -1,5 +1,40 @@
...
+                            {link.icon}
+                        </a>
+                    ))}
+                    <Mailto email="alexandra.rychka@gmail.com"></Mailto>
```
Comment (kusochek) [inline src/components/footer/Footer.jsx]: Avoid to use closing tag if you don’t use anything inside tag

File: src/components/list/List.jsx [hunk 4]
```diff
@@ -1,5 +1,79 @@
...
+import { Link } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { changeList } from '../../store/list/listSlice';
+import useListQueries from './listHooks';
+import { fetchRecipe } from '../../store/recipe/recipeSlice';
+
 function List() {
-    return <section>List</section>;
+    const curCuisine = useSelector((state) => state.list?.cuisine);
```
Comment (kusochek) [inline src/components/list/List.jsx]: { list } here better to use destruction 
And in the lower case as well

File: src/components/list/List.jsx [hunk 5]
```diff
@@ -1,5 +1,79 @@
...
+
+    const { data, error, isLoading, refetch } = useListQueries();
+
+    useEffect(() => {
+        if (!curCuisine && !search) {
+            refetch();
+        }
+    }, [refetch, curCuisine, search]);
+
+    const dispatch = useDispatch();
```
Comment (kusochek) [inline src/components/list/List.jsx]: This logic should be on the top of component

File: src/components/list/List.jsx [hunk 6]
```diff
@@ -1,5 +1,79 @@
...
+    const openRecipe = (recipeId) => {
+        dispatch(fetchRecipe({ id: recipeId }));
+    };
+
+    const buttonClicked = () => {
```
Comment (kusochek) [inline src/components/list/List.jsx]: Name this function more clear

File: src/components/cuisines/Cuisines.jsx [hunk 7]
```diff
@@ -1,5 +1,38 @@
...
+        dispatch(
+            changeList({
+                cuisine: cuisine,
+                searchParam: '',
+            })
```
Comment (kusochek) [inline src/components/cuisines/Cuisines.jsx]: move this logic in the variable 
const updatedCuisines = {
                cuisine: cuisine,
                searchParam: '',
            }

File: src/components/list/List.jsx [hunk 8]
```diff
@@ -1,5 +1,79 @@
...
+            search
+                ? changeList({
+                      cuisine: '',
+                      offset: offsetNum + 15,
+                  })
+                : changeList({
+                      cuisine: curCuisine,
+                      searchParam: '',
+                      offset: offsetNum + 15,
+                  })
```
Comment (kusochek) [inline src/components/list/List.jsx]: here better to create let with default object what you should pass to changeList
than write if (search)... and add this field and after call changeList

File: src/components/list/listHooks.js [hunk 9]
```diff
@@ -1 +1,109 @@
...
+import {
+    useGetRandomRecipesQuery,
+    useGetRecipesByCuisineQuery,
+    useSearchRecipesQuery,
+} from '../../api/api.js';
+
+function useListQueries() {
+    const cuisine = useSelector((state) => state.list?.cuisine);
+    const searchParam = useSelector((state) => state.list?.searchParam);
+    const offset = useSelector((state) => state.list.offset);
```
Comment (kusochek) [inline src/components/list/listHooks.js]: use destracture

File: src/components/list/listHooks.js [hunk 10]
```diff
@@ -1 +1,109 @@
...
+    //loading the recipes according to the search
+    const {
+        data: listBySearch,
+        error: errorBySearch,
+        isLoading: isLoadingBySearch,
+    } = useSearchRecipesQuery(
+        { searchParam, offset },
+        {
+            skip: !searchParam,
+        }
```
Comment (kusochek) [inline src/components/list/listHooks.js]: write it in one line

File: src/components/list/listHooks.js
Comment (kusochek) [inline src/components/list/listHooks.js:1]: too much useEffect here, need to rewrite this component
Let's write here comment
// TODO: rewrite this hook without useEffects

File: src/components/footer/Footer.jsx (hunk 2 above)
Comment (sashadmii) [reply src/components/footer/Footer.jsx:29]: can't really do much here, my code editor saves this as it is(

File: src/components/list/List.jsx (hunk 4 above)
Comment (sashadmii) [reply src/components/list/List.jsx]: this is a typo, it shouldn't be here

File: src/components/cuisines/Cuisines.jsx (hunk 7 above)
Comment (sashadmii) [reply src/components/cuisines/Cuisines.jsx]: not sure how to do this, I'm passing cuisine param, when dispatching the action;
how should I pas it to the variable?

const fetchCuisine = (cuisine) => {
        dispatch(
            changeList({
                cuisine: cuisine,
                searchParam: '',
            })
        );
    };

File: src/components/list/List.jsx (hunk 8 above)
Comment (sashadmii) [reply src/components/list/List.jsx]: also, not really getting the idea of how to achieve this

**Pull request merged**